### PR TITLE
Add --read-only-tmpfs options

### PR DESCRIPTION
**Pull request by rhatdan**
The --read-only-tmpfs option caused podman to mount tmpfs on /run, /tmp, /var/tmp
if the container is running int read-only mode.

The default is true, so you would need to execute a command like

--read-only --read-only-tmpfs=false to turn off this behaviour.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

File: cmd/podman/common.go [hunk 1]
```diff
@@ -434,6 +434,10 @@ func getCreateFlags(c *cliconfig.PodmanCommand) {
 		"read-only", false,
 		"Make containers root filesystem read-only",
 	)
+	createFlags.Bool(
+		"read-only-tmpfs", true,
+		"When running containers in read-only mode mount tmpfs on /run, /tmp and /var/tmp",
```
Comment (baude) [inline cmd/podman/common.go]: mount tmpfs on /run, /tmp and /var/tmp as read-only ?

**Comment by baude** [comment]
just the one nit, thanks for doing the remote stuff as well!

lgtm

File: cmd/podman/common.go (hunk 1 above)
Comment (jamescassell) [reply cmd/podman/common.go]:
```suggestion
		"When running containers in read-only mode mount a read-write tmpfs on /run, /tmp and /var/tmp",
```

File: docs/podman-create.1.md [hunk 2]
```diff
@@ -542,6 +542,9 @@ By default a container will have its root filesystem writable allowing processes
 to write files anywhere.  By specifying the `--read-only` flag the container will have
 its root filesystem mounted as read only prohibiting any writes.
 
+**--read-only-tmpfs**=*true*|*false*
+If container is running in --read-only mode, then mount a tmpfs on /run, /tmp, and /var/tmp.  The default is *true*
```
Comment (TomSweeneyRedHat) [inline docs/podman-create.1.md]: If you adjust common.go, adjust here too please.

File: docs/podman-run.1.md [hunk 3]
```diff
@@ -534,6 +534,9 @@ By default a container will have its root filesystem writable allowing processes
 to write files anywhere.  By specifying the `--read-only` flag the container will have
 its root filesystem mounted as read only prohibiting any writes.
 
+**--read-only-tmpfs**=*true*|*false*
+If container is running in --read-only mode, then mount a tmpfs on /run, /tmp, and /var/tmp.  The default is *true*
+
```
Comment (TomSweeneyRedHat) [inline docs/podman-run.1.md]: ditto message change.

**Comment by rhatdan** [comment]
@baude @TomSweeneyRedHat @mheon @vrothberg @giuseppe PTAL

**Comment by giuseppe** [comment]
LGTM

**Comment by rhatdan** [comment]
@TomSweeneyRedHat I made your message change could you or someone else give me the magical /lgtm

**Pull request merged**